pulp_docker/tests/functional/api/test_recursive_remove.py

```diff
@@ -0,0 +1,333 @@
+# # coding=utf-8
+"""Tests that recursively remove docker content from repositories."""
+import unittest
+
+from pulp_smash import api, config
+from pulp_smash.pulp3.constants import REPO_PATH
+from pulp_smash.pulp3.utils import gen_repo, sync
+from requests.exceptions import HTTPError
+
+from pulp_docker.tests.functional.constants import (
+    DOCKER_TAG_PATH,
+    DOCKER_REMOTE_PATH,
+    DOCKER_RECURSIVE_ADD_PATH,
+    DOCKER_RECURSIVE_REMOVE_PATH,
+    DOCKERHUB_PULP_FIXTURE_1,
+)
+from pulp_docker.tests.functional.utils import gen_docker_remote
+
+
+class TestRecursiveRemove(unittest.TestCase):
```

Comment:
Suggestion: add a test that addresses my previous comment regarding "in repo version" content, e.g.  initial repo version contains Manifest M -> Blob B. Recursively remove a Tag T -> M (which is not part of the repo version). Verify that the new repo version still contains M -> B.